app/controllers/collections_controller.rb

```diff
@@ -118,6 +123,17 @@ def leave
     end
   end
 
+  def toggle_favorite
```

Comment:
I'm more a fan of individual controllers following RESTful resource style, e.g. `POST /favorites` here.
But you have if different style in this project anyway.

Reply:
I see what you mean, but would still keep it, since it matches the current style as you said.